app/src/main/kotlin/com/yubico/yubioath/ui/main/CredentialFragment.kt

```diff
@@ -214,7 +213,7 @@ class CredentialFragment : ListFragment(), CoroutineScope {
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
         super.onActivityResult(requestCode, resultCode, data)
 
-        fun handleUrlCredential(context: Context, data: String) {
+        fun handleUrlCredential(context: Context, data: String?) {
```

Comment:
if your data nullable than you need to check for null, otherwise you will crash on uri parse

Reply:
I guessed it would fail inside the parsing if that was the case, and the try catch block would take care of that, but yes, it is safer to handle the NPE manually. Thanks for pointing it out. Fixed now.